### ISSUE TITLE
Is it possible to get metadata from a prepared statement

### ISSUE THREAD

**Issue by wcn00** [denisenkom/go-mssqldb#567]
I need to extract column and param info from a prepared statement to lay out a GUI.  In this driver is there any way to do that?  

I do see that the prepared statement network buffer is there (but very private).  Is there any doc for that buffer?
Thanks in advance
wcn

**Comment by kardianos** [comment]
I'm not sure what data from a prepared statement would give you. Prepared statements aren't like Postgresql. You actually don't need to prepare a statment. Typically in sql server you run a special mode of a command to execute a command without results / side-effects.

**Comment by wcn00** [comment]
That would be something like a query explain?  I will research that 
idea.  Let me know if you have any other pointers and thanks for the 
response.

wcn

**Comment by tc-hib** [comment]
Hello, you might want to have a look at [sp_describe_first_resultset](https://docs.microsoft.com/fr-fr/sql/relational-databases/system-stored-procedures/sp-describe-first-result-set-transact-sql?view=sql-server-ver15).
There's also [sp_describe_undeclared_parameters](https://docs.microsoft.com/fr-fr/sql/relational-databases/system-stored-procedures/sp-describe-undeclared-parameters-transact-sql?view=sql-server-ver15).

**Comment by wcn00** [comment]
Is it possible to run tsql statements in via the golang driver?

wcn

**Comment by kardianos** [comment]
Yes. In fact, that's all that is possible to run. T-SQL is the one (and
only) SQL dialect that SQL Server accepts (well, technically it knows an
ODBC flavor as well).
Unlike postgresql, when you run SQL, it is T-SQL by default, you don't need
to "switch into" T-SQL.

**Comment by wcn00** [comment]
Thanks TC and Daniel, I think I can make this work ;)

wcn
